Bug fix: Warns user if incorrect SDK version is installed

EndlessTrax opened the pull request:
##### SUMMARY

This raises an error if the wrong `python-dsv-sdk` version is installed. This lookup plugin currently only supports v0.0.1 of the SDK. This is a temporary fix until full support with the latest SDK version can be developed.

(Re-submitted with clean git history)

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

dsv lookup plugin

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
Before the fix, the error presented to the user:
```
Error was a <class 'ansible.errors.AnsibleError'>, original message: An unhandled exception occurred while running the lookup plugin 'community.general.dsv'. Error was a <class 'TypeError'>, original message: __init__() got an unexpected keyword argument 'tenant'"}
```
After:
```
Error was a <class 'ansible.errors.AnsibleError'>, original message: python-dsv-sdk==0.0.1 must be installed to use this plugin"}
```

felixfontein commented on plugins/lookup/dsv.py:
```diff
@@ -105,11 +105,15 @@
+        try: 
+            vault = SecretsVault(**vault_parameters)
+            return vault
+        except:
```
Better be more specific here:
```suggestion
        except TypeError:
```

felixfontein reviewed (commented):
Thanks for your contribution! Can you add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

EndlessTrax commented:
> Thanks for your contribution! Can you add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Added. Also changed the bare exception as suggested. Thanks for the review.

felixfontein commented on changelogs/fragments/4422-warn-user-if-incorrect-SDK-version-is-installed.yaml:
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - dsv_lookup - Raise an Ansible error if the wrong `python-dsv-sdk` version is installed.
```
```suggestion
  - dsv lookup plugin - raise an Ansible error if the wrong ``python-dsv-sdk`` version is installed (https://github.com/ansible-collections/community.general/pull/4422).
```

felixfontein commented:
@EndlessTrax thanks for fixing this!
